packages/health/android/src/main/kotlin/cachet/plugins/health/HealthPlugin.kt

```diff
@@ -599,6 +602,26 @@ class HealthPlugin(private var channel: MethodChannel? = null) : MethodCallHandl
           .addOnSuccessListener(threadPoolExecutor!!, workoutDataHandler(type, result))
           .addOnFailureListener(errHandler(result))
       }
+      /*
+      BMR is not available to retrieve for every day (although it is reflected in Google Fit energy calcs for every single day)
+      Thus, a specific case is defined below for handling BMR, with "1" provided as startDate to the setTimeRange method.
+      We have also set a limit of "1" through setLimit, to only get the latest value.
+      Note:
+        According to Google documentation, since "the recorded BMR is instantaneous, the start time should not be set.".
+        However, startTime is oddly a required property. 
+      */
+      DataType.TYPE_BASAL_METABOLIC_RATE -> {
+        Fitness.getHistoryClient(activity!!.applicationContext, googleSignInAccount)
+          .readData(
+            DataReadRequest.Builder()
+              .read(dataType)
+              .setTimeRange(1, endTime, TimeUnit.MILLISECONDS)
+              .setLimit(1)
+              .build()
```

Comment:
I wanted to say, there are some nice Kotlin things you can do to make this more readable/expressive (e.g. `Builder().apply { ... } .build()`), but . . . this entire file has much bigger problems :cry:

Reply:
Yeah, we're trying to keep our changes to minimum, to hopefully be able to have them merged into the official repo. Have already got two other super small PRs there, without a response though.